tests/gtest/test_utils.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef TEST_UTILS_H
+#define TEST_UTILS_H
+
+#include "cppapi.hpp"
+#include "ctypes.h"
+#include <iostream>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+const std::string CREDENTIALS = "";
+static constexpr float fill = -999.25;
+
+class DatahandleCubeIntersectionTest : public ::testing::Test {
```

Comment:
Should we make separate classes for each type of handle? I think that would make the tests more readable.

I think all the methods you define on this class in later commits could be free functions.

Reply:
How would defining separate class for SingleDatahandleCubeIntersectionTest and DoubleReverseDatahandleCubeIntersectionTest and DoubleDifferentSizeDatahandleCubeIntersectionTest make tests more readable?

The tests would differ only in class name, and I would have to either repeat the same information in every class (which I don't like) or create a parent class with all this information (which we established you don't like).

Now one has to look at the datahandle name to figure out what it represents, otherwise you'll have to look at the class name. I'd say variable name would contain less information, thus easier to process than class name.
So I see zero benefit. For now defining class by test category is totally fine. 

Regarding free functions: then I will have to either rename these functions or add namespaces, as for the new class for tests based on 10 samples testdata the logic would be the same - I would also check attributes, slices and fences.
Encapsulation of logic in the class is more straightforward to me.

Reply:
I don't have anything against inheritance in general, only when it ties together code that should not necessarily be tied together.

If you prefer to keep it as is I'm okay with that.